### ISSUE TITLE
(evaluate, GateSystem, Matrix) at polynomials with complex coefficients fails

### ISSUE THREAD

**Issue by timduff35** [Macaulay2/M2#2545]
```
needsPackage "SLPexpressions";
needsPackage "NumericalAlgebraicGeometry";
declareVariable x;
G = gateSystem(gateMatrix{{x}},gateMatrix{{x^2-x-1}});
R = CC[y];
evaluate(G, vars R)
```
throws an error 
```
i6 : evaluate(G, vars R)
stdio:6:1:(3): error: no method found for applying raw to:
     argument   :  RR (of class InexactFieldFamily)
```
@antonleykin, didn't this used to work? By the way, I don't see any error if CC is replaced by RR, or QQ.

**Comment by antonleykin** [comment]
It looks like something is wrong with promotion, which is unrelated to the packages.

`R = CC[x]; matrix 0 | vars R` produces your error, while (as you point out) `R = RR[x]; matrix 0 | vars R` does not.

**Comment by DanGrayson** [comment]
Due to this, no doubt:
```m2
i1 : promote ( matrix 0 , CC_53[x] )
stdio:1:1:(3): error: no method found for applying raw to:
     argument   :  RR (of class InexactFieldFamily)
```

**Comment by DanGrayson** [comment]
RR should not be in this list, it should be RR_53 instead:

```m2

i1 : (CC[x]).baseRings

o1 = {ZZ, QQ, RR, CC  }
                    53

o1 : List
```

**Comment by DanGrayson** [comment]
Same here:
```m2

i4 : (CC_53).baseRings

o4 = {ZZ, QQ, RR}

o4 : List

i4 : (CC_53).baseRings

o4 = {ZZ, QQ, RR}

o4 : List
```

**Comment by DanGrayson** [comment]
fixed here: commit 7dcb1035ff984078addf6e7da8d09f436cfe19e4
